detector.go

```diff
@@ -62,16 +102,18 @@ type DetectConfig struct {
 }
 
 func (c *DetectConfig) process(done []Buildpack) ([]Buildpack, []BuildPlanEntry, error) {
-	var runs []detectRun
+	var runs []DetectRun
 	for _, bp := range done {
 		t, ok := c.runs.Load(bp.String())
 		if !ok {
 			return nil, nil, errors.Errorf("missing detection of '%s'", bp)
 		}
-		run := t.(detectRun)
+		run := t.(DetectRun)
 		outputLogf := c.Logger.Debugf
 
 		switch run.Code {
+		case CodeVersionFail:
```

Comment:
Instead of returning here I think we should be in the `default:` case so that this is fatal to group detection for required buildpacks but not for optional buildpacks